File: 0141-linked-list-cycle/0141-linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
 function hasCycle(head) {
    let jumper = head;

    while(jumper && jumper.next){
        head = head.next;
        jumper = jumper.next.next;
        if(head === jumper) return true;
    }
    return false;
 }

File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        cur = head

        while cur and cur.next:
            head = head.next
            cur = cur.next.next

            if head == cur:
                return True
        return False